api/src/retry.rs

```diff
@@ -152,12 +152,12 @@ mod tests {
         ok.assert();
 
         // Retries up to N times on timeout for non-first-requests.
-        for i in 0..10 {
+        for retry_count in 0..10 {
```

Comment:
I'm all right with `retry_count` here, but in general I do like the `i_retry` convention. `retry_count` is ambiguous: is this the "max number of retries" or the "current retry number". `i_retry` and `num_retries` is what we tend to default to, but I think there's more variability here in the team than with other things.